src/main/java/net/minestom/server/listener/ChatMessageListener.java

```diff
@@ -40,30 +34,20 @@ public static void chatMessageListener(ClientChatMessagePacket packet, Player pl
         }
 
         final Collection<Player> players = CONNECTION_MANAGER.getOnlinePlayers();
-        PlayerChatEvent playerChatEvent = new PlayerChatEvent(player, players, (e) -> buildDefaultChatMessage(e.getPlayer(), e.getMessage()), message);
+        PlayerChatEvent playerChatEvent = new PlayerChatEvent(player, players, message);
 
         // Call the event
         EventDispatcher.callCancellable(playerChatEvent, () -> {
-            final Function<PlayerChatEvent, Component> formatFunction = playerChatEvent.getChatFormatFunction();
-            Component textObject = formatFunction.apply(playerChatEvent);
-
             final Collection<Player> recipients = playerChatEvent.getRecipients();
+
             if (!recipients.isEmpty()) {
                 // delegate to the messenger to avoid sending messages we shouldn't be
-                Messenger.sendMessage(recipients, textObject, ChatPosition.CHAT, player.getUuid());
+                Messenger.sendMessage(
+                        recipients,
+                        playerChatEvent.getFinalMessage(),
+                        ChatPosition.CHAT,
+                        player.getUuid());
             }
         });
     }
-
-    private static @NotNull Component buildDefaultChatMessage(@NotNull Player player, @NotNull String message) {
-        final String username = player.getUsername();
-        return Component.translatable("chat.type.text")
-                .args(Component.text(username)
-                                .insertion(username)
-                                .clickEvent(ClickEvent.suggestCommand("/msg " + username + " "))
```

Comment:
I don't really understand why you remove the clickEvent line here. Can you tell me the reason?

Reply:
The command is not implemented in minestom and it might not be the desired behavior